src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -377,11 +378,13 @@ internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diag
                 SeverityFilter severityFilter,
                 Compilation? compilation,
                 AnalyzerOptions? analyzerOptions,
+                AnalysisScope? analysisScope,
                 CancellationToken cancellationToken)
             {
                 if (compilation != null && compilation.Options.SyntaxTreeOptionsProvider is { } treeOptions)
                 {
-                    foreach (var tree in compilation.SyntaxTrees)
+                    var trees = analysisScope?.SyntaxTrees ?? compilation.SyntaxTrees;
```

Comment:
Null checks for `compilation` and `analysisScope` can be removed in this method once #67591 is addressed